### ISSUE TITLE
React to --features in pre-build

### ISSUE THREAD

**Issue by adamspofford-dfinity** [cross-rs/cross#1275]
### Checklist

- [X] I've looked through the [issues and pull requests](https://github.com/cross-rs/cross/issues?q=) for similar request
- [ ] This feature could be solved with a [custom docker image](https://github.com/cross-rs/cross#custom-docker-images) (optional)

### Describe your request

There does not seem to be a way to react to optional crate features in the pre-build script or a custom Dockerfile. It would be great if an environment variable or docker arg or similar that reflected the crate's selected features.

### Describe why this would be a good inclusion for `cross`

For example, avoiding installing/building a large library that is required for an optional feature if the feature is disabled.

**Comment by Emilgardis** [comment]
I agree, this might be hard to do however, maybe we can use the cargo metadata we already gather.

This is kinda related to https://github.com/cross-rs/cross/issues/1244 but for the custom build
